### DISCUSSION TITLE
How to submit a workflow with input parameter by Argo python SDK

### DISCUSSION THREAD

**Discussion by pockyhe** [argoproj/argo-workflows#8897] [Q&A] [answered]
Hello, I wanna use Argo python SDK to submit workflow with input parameter. I don't know where to inject parameter. Could you help me with any suggestion?
![image](https://user-images.githubusercontent.com/64390829/171584004-45f26aa4-a37a-47e4-89c5-9c02550f1443.png)

like the image above, there is input parameter: message, I need to inject while submitting workflow

**Comment by terrytangyuan** [comment, marked as answer]
Instead of loading your workflow spec from YAML, you could construct it via Python:

```
manifest = IoArgoprojWorkflowV1alpha1Workflow(
    metadata=ObjectMeta(generate_name='hello-world-'),
    spec=IoArgoprojWorkflowV1alpha1WorkflowSpec(
        entrypoint='whalesay',
        templates=[
            IoArgoprojWorkflowV1alpha1Template(
                name='whalesay',
                inputs=IoArgoprojWorkflowV1alpha1Inputs(...)
                container=Container(
                    image='docker/whalesay:latest', command=['cowsay'], args=['hello world']))]))

```

**Reply by pockyhe** [reply]
so there is no way to directly inject parameter into workflow with API?

**Reply by terrytangyuan** [reply]
You can also modify the constructed object directly.

**Reply by Smile-Autra** [reply]
I don't think is a good solution. For we want to seperate the workflow yaml with the submit logics. Is there any alternative method for fill the parameter use SDK？
